lib/min_heap.rb

```diff
@@ -8,24 +8,39 @@ def initialize(key, value)
 end
 
 class MinHeap
+  attr_reader :store
 
   def initialize
     @store = []
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(logn)
+  # Space Complexity: O(1)
   def add(key, value = key)
-    raise NotImplementedError, "Method not implemented yet..."
+    if @store.empty?
+      @store << HeapNode.new(key, value)
+    else
+      new_i = @store.length
+      @store[new_i] = HeapNode.new(key, value)
+      heap_up(new_i)
+    end
+    return key
   end
 
   # This method removes and returns an element from the heap
   #   maintaining the heap structure
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(logn)
+  # Space Complexity: O(1)
   def remove()
-    raise NotImplementedError, "Method not implemented yet..."
+    min = @store.min_by {|node| node.key}
```

Comment:
This is a **linear search**, you're totally losing the advantage of a Heap here!